BFS/src/Bridge.java

```diff
@@ -0,0 +1,69 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.File;
+
+public class Bridge {
+
+    private static int edge_num, start, goal;
+    private static ArrayList<Integer> x_vals, y_vals;
+
+    private static void start(String file) {
+
+        try {
+            Scanner s = new Scanner(new File(file));
+            s.nextInt();
+            edge_num = s.nextInt();
+            start = s.nextInt();
+            goal = s.nextInt();
+            
+            x_vals = new ArrayList<Integer>();
+            y_vals = new ArrayList<Integer>();
+            
+            for (int i = 0; i < edge_num; i++) {
+                x_vals.add(s.nextInt());
+                y_vals.add(s.nextInt());
+            }
+       
+        } catch (Exception e) {
+            System.out.println("file not opened");
```

Comment:
instead of having this, you can just throw in a throw method in main: 
`public static void main(String[] args) throws FileNotFoundException `